app/src/main/java/com/example/habitsmasher/ui/dashboard/DashboardFragment.java

```diff
@@ -73,8 +99,17 @@ public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction)
         }
     };
 
+
+    /**
+     * Adds a new habit to the habit list.
+     */
+    @Override
+    public void addNewHabit(String title, String reason, Date date) {
+        Habit newHabit = new Habit(title, reason, date);
+        _habits.add(newHabit);
+    }
```

Comment:
I think we should be doing the firebase stuff **here**. We should contain all "adding habit" functionality to the add habit method. We can also unit test this as well.